### PR TITLE
fluids: Fix post-processing in DMPlexInsertBoundaryValues_NS

### PR DESCRIPTION
**Pull request by jrwrigh**
From this comment: https://github.com/CEED/libCEED/commit/b80a4419c8f00e2f8e9a7afd897117055f7f7a73#r76899317

Basically don't zero the full `Q_loc` vector so that it can be correctly used by `VecView`.

### CODE REVIEW

File: examples/fluids/src/misc.c [hunk 1]
```diff
@@ -98,18 +98,21 @@ PetscErrorCode ICs_FixMultiplicity(DM dm, CeedData ceed_data, User user,
   PetscFunctionReturn(0);
 }
 
-
 // Note: The BCs must be inserted *before* the other values are inserted into Q_loc
```
Comment (jeremylt) [inline examples/fluids/src/misc.c:101]: We can delete this note now

**Review by LeilaGhaffari — APPROVED** [review]
LGTM! Thanks for the fix!

File: examples/fluids/src/misc.c [hunk 2]
```diff
@@ -260,6 +264,18 @@ PetscErrorCode SetBCsFromICs_NS(DM dm, Vec Q, Vec Q_loc) {
                                     "DMPlexInsertBoundaryValues_C", DMPlexInsertBoundaryValues_NS);
   CHKERRQ(ierr);
 
+  PetscCall(DMHasNamedLocalVector(dm, "boundary mask", &boundary_mask_created));
+  if (!boundary_mask_created) {
```
Comment (jeremylt) [inline examples/fluids/src/misc.c:268]: We may have to think a bit more about this - In what situation would this mask already exist and be valid for this boundary configuration?

Comment (jrwrigh) [reply examples/fluids/src/misc.c:268]: I mainly put it there for the dirichlet branch, which currently creates it in a different location. Though they shouldn't be called at the same time and `SetBCsFromICs_NS` would overwrite `DMPlexInsertBoundaryValues` from what the dirichlet function would write anyways.

Comment (jrwrigh) [reply examples/fluids/src/misc.c:268]: Arguably `SetBCsFromICs` should either be merged in with the Dirichlet function when that gets merged in anyways.

Comment (jrwrigh) [reply examples/fluids/src/misc.c:268]: Regardless, I'll go ahead and remove that line. I was being a bit over-cautious I think.

File: examples/fluids/src/misc.c (hunk 1 above)
Comment (jrwrigh) [reply examples/fluids/src/misc.c]: Done

File: examples/fluids/src/misc.c (hunk 2 above)
Comment (jrwrigh) [reply examples/fluids/src/misc.c]: Removed the if statement.

**Review by jeremylt — APPROVED** [review]
LGTM, thanks

**Comment by jedbrown** [comment]
Thanks!

**Pull request merged**